contents/swift/welcome-to-swift.md

```diff
@@ -102,6 +128,29 @@ You can think of Structs as a way to create instances that have their own unique
 
 If you wish to find out more, here is an [article](https://medium.com/capital-one-developers/reference-and-value-types-in-swift-de792db330b2) that explains the difference between the 2 types, as well as the benefits of value types and when to use them.
 
+## Enum
+
+Swift's enums can have associated values. For example:
+
+```swift
+enum Barcode {
+    case upc(Int, Int, Int, Int)
+    case qrCode(String)
+}
+
+let productBarcode = Barcode.qrCode("foo")
+
+switch productBarcode {
```

Comment:
switch for enums in Swift must be exhaustive. might be good to mention this magic fact (Xcode would complain this and not compile)